src/components/app.tsx

```diff
@@ -10,8 +10,43 @@ import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
+import {
+  ActionHandler,
+  DatasetAsyncAction,
+  datasetUrlLoad,
+} from '../actions';
+
+
+
+interface DataURL {
+  name: string;
+  url: string;
+};
+export type VoyagerData = DataURL | Object[];
```

Comment:
Also, wonder if there is any reason to diverge from `Data` object in Vega-Lite yet.  

If not, we could use consistent structure and use `isUrlData`, `isInlineData` type guards provided in `vega-lite/src/data`